lib/server.js

```diff
@@ -24,6 +26,16 @@ const initServer = function(
     applications = apps.createAppsIndex(applications);
   }
 
+  // versions cached for efficient search when provided version is a range
+  this._cachedVersions = new Map();
+  applications.forEach((appVersions, appName) => {
+    const versions = Array.from(appVersions)
+      .filter(app => app[0] !== 'latest')
+      .map(version => [new semver.SemVer(version[0]), version[1]])
```

Comment:
@belochub @aqrln Also I see no particular reason to specifically emphasize arguments with destructuring here and it should be sufficient to rename `.filter` argument to `version` to be consistent, bit I don't really care so it's up to you.